sys/luid/luid.c

```diff
@@ -26,7 +26,39 @@
 
 #include "luid.h"
 
-static uint8_t lastused = 1;
+#if CPUID_LEN
+void __attribute__((weak)) luid_base(void *buf, size_t len)
+{
+    uint8_t *out = (uint8_t *)buf;
+    uint8_t cid[CPUID_LEN];
+
+    cpuid_get(cid);
```

Comment:
Why did you remove the setting of the seed in this version?

Reply:
I initially changed that when I wanted to just overwrite `luid_base()`, can restore the old behaviour.
But I don't see much value in xoring the CPU ID with 0x23 - why not just use it as is?

Reply:
Well, the seed is overridable at compile time. Also if it ensures that `out` is changed when (in the old behavior) the length of the CPU ID is shorter than `len`.

Reply:
BTW, I would still prefer if _all_ bytes of the CPU ID would be used. As we can't make assumptions on the CPU ID, taking just the first `len` bytes might lead to just the non-changing part to be read.

Reply:
(I wouldn't expect to be the LSB be the changing part either. I remember, that when implementing this we observed also CPU IDs that changed in the middle part)

Reply:
The formats seem to be pretty crazy. I could find some details documentation on STM32: The first byte is an 8 bit counter of the wafer, the four last bytes are x and y coordinate of the MCU in that wafer as binary coded decimals. The middle part is the lot identifier as ASCII. I bet this is basically a device identifier with a counter in it to increment whenever the the wafer counter overflows...

So a big +1 for not assuming which part is the "valuable" part of the CPU ID. And also it could be pretty much distributed, e.g. due to the bcd encoding some bits of the last part of the STM32 CPU ID is basically dead, even though the wafer-x and wafer-y position is likely the one yielding the biggest hamming distance between two randomly selected STM32 chips...